src/components/textToSpeech/js/TextToSpeech.js

```diff
@@ -170,7 +242,11 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
             paused: false
         };
 
-        if (!that.queue.length) {
+        if (that.queue.currentUtterance) {
+            that.queue.currentUtterance = undefined;
```

Comment:
I actually meant to put it in the queue item, rather than to rebase the entire queue structure.
So, on line 280 we would have that.queue.push({text: text, utterance: toSpeak})
queue can then just go back to being a [] and the rest of the changes reverted